odoo/fields.py

```diff
@@ -2564,6 +2564,13 @@ def write(self, records, value):
         dirty = self.column_type and self.store and any(records._ids)
         records.env.cache.update(records, self, itertools.repeat(cache_value), dirty=dirty)
 
+    def _inverse_related(self, records):
+        super()._inverse_related(records)
+        if self.max_width or self.max_height:
+            # Invalidate field because the _image_process isn't done on the cache during
+            # write or create (see Image.create/write)
+            records.invalidate_recordset([self.name])
```

Comment:
This doesn't look correct to me.  Consider the field `self` to be `store=True, attachment=False`.  This call to `invalidate_recordset()` will flush the field to the database before invalidating the cache :upside_down_face: 
In this case, I suspect the best is to recompute the field, and store the expected (resized) value in cache.